urldee901fae6_DocX_personal_timetabling_tgz-pJ8-net_personaltt_main_BenchmarkJ8/src/net/personaltt/main/Benchmark.java

```diff
@@ -0,0 +1,789 @@
+        System.out.printf("PROBLEM S - %s number\n", number);
+        ProblemDefinition p = bench5(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void testP(int number) {
+        System.out.printf("PROBLEM P - %s number\n", number);
+        ProblemDefinition p = benchPriority(1,number);
+        printInitialCosts(p);
+        testProblem(p);
+    }    
+
+    private static void printInitialCosts(ProblemDefinition p) {
+           
+        MainSolverState currentSolution = new MainSolverState();
+        currentSolution.init(p.initialSchedule);
+        
+        System.out.println(String.format("INITIAL COST: c:%s p:%s", currentSolution.constraintsCost(), currentSolution.preferenceCost()));
+    }
+    
+ 
+    
+    
+    public static void benchmarkPriority() {
+        List<String> results = new ArrayList<String>();
+        for (int i = 0; i <= 0; i+=1) {
+            int nonPrefCount = 0;
+            for (int j = 0; j < 10; j++) {
+                ProblemDefinition problem = benchPriority(i,50);
+                SolverState state = benchSolve(problem);
+                if (state.getBestSolution().getSchedule().getAllocationOf(1).getStart() != 0) {
+                    nonPrefCount++;
+                }
+            }
+            results.add(String.format("Priority: %s, not preffered count: %s/10", i, nonPrefCount));
+        }
+       
+        for (String string : results) {
+            System.out.println(string);
+        }
+    }
+    
+    /**
+     * Experimental benchmark for determination relation between
+     * number of events and interation in whish is found best solution
+     */
+    public static void benchmarkTerminationCondition() {
+        List<String> results = new ArrayList<String>();
+        for (int j = 100; j <= 1000; j+=100) {
+            ProblemDefinition problem = bench1(j);
+            SolverState state = benchSolve(problem);
+            results.add(String.format("Events: %s, best iteration: %s, end iteration: %s",
+                    j, 
+                    state.getLastBestIteration(),
+                    state.getItearation()
+                    ));
+        }
+       
+        for (String string : results) {
+            System.out.println(string);
+        }
+    }
+    
+    public static void benchmarkProp(ProblemDefinition problem, String prop, double min, double max, double step) {
+        
+        List<String> results = new ArrayList<>();
+        
+        for (double stay = min; stay <= max; stay+= step) {
+            long ccost = 0;
+            long pcost = 0;
+            long besti = 0;
+            for (int i = 0; i < 5; i++) {
+                solverProperties.setProperty(prop, String.valueOf(stay));
+                
+                SolverState state = benchSolve(problem);
+                results.add(
+                        String.format("test prop:%s, i:%s result: best c:%s p:%s it:%s",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getLastBestIteration&#39; expression, i (declared at line 224). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 229. Consider formatting this data with &#39;%d&#39; instead.